src/ocean/meta/types/Function.d

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+
+    Traits specific to various function-like types
+
+    NB: because this module is often used as purely compile-time dependency it
+        used built-in asserts instead of `ocean.core.Test` to reduce amount of
+        cyclic imports. `ocean.meta` modules in general are not supposed to
+        import anything outside of `ocean.meta`.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.types.Function;
+
+import ocean.meta.AliasSeq;
+import ocean.meta.traits.Basic;
+import ocean.meta.types.Typedef;
+
+/*******************************************************************************
+
+    Evaluates to a compile-time sequence representing the parameters of Fn
+
+    Params:
+        Callable = callable type (function/delegate/function pointer) or
+            function symbol
+
+*******************************************************************************/
+
+public template ParametersOf ( Callable... )
+{
+    static assert (Callable.length == 1);
```

Comment:
Why is it a variadic template and not `ParametersOf ( Callable )`?

Reply:
Ah, it is probably to allow for accepting a function symbol as a template argument. Still this needs a brief note to avoid confusion.